perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/TasksManagerBlImpl.java

```diff
@@ -361,85 +270,124 @@ public TaskResult getTaskResultById(int taskResultId) {
+	}
 
-			// fill service state
-			serviceState.setTask(task);
-			serviceStates.get(taskService).setBlockedOnFacility(getServicesManagerBl().isServiceBlockedOnFacility(task.getService(), facility));
+	@Override
+	public List<TaskResult> getTaskResultsByDestinations(PerunSession session, List<String> destinationsNames) {
+		return getTasksManagerImpl().getTaskResultsByDestinations(destinationsNames);
+	}
 
-		}
+	public TasksManagerImplApi getTasksManagerImpl() {
+		return tasksManagerImpl;
+	}
 
-		return new ArrayList<>(serviceStates.values());
+	@Override
+	public int insertNewTaskResult(PerunSession sess, TaskResult taskResult) {
+		return getTasksManagerImpl().insertNewTaskResult(taskResult);
+	}
 
+	@Override
+	public int insertTask(PerunSession sess, Task task) {
+		return getTasksManagerImpl().insertTask(task);
 	}
 
 	@Override
-	public void deleteTask(PerunSession sess, Task task) {
+	public boolean isThereSuchTask(PerunSession sess, Service service, Facility facility) {
+		return getTasksManagerImpl().isThereSuchTask(service, facility);
+	}
 
-		Facility facility = task.getFacility();
-		Service service = task.getService();
+	@Override
+	public List<Task> listAllTasks(PerunSession perunSession) {
+		return getTasksManagerImpl().listAllTasks();
+	}
 
-		// clear all task results
-		getTasksManagerImpl().deleteTaskResults(task.getId());
+	@Override
+	public List<Task> listAllTasksForFacility(PerunSession session, int facilityId) {
+		return getTasksManagerImpl().listAllTasksForFacility(facilityId);
+	}
 
-		// remove task itself
-		getTasksManagerImpl().removeTask(service, facility);
+	@Override
+	public List<Task> listAllTasksForService(PerunSession sess, int serviceId) {
+		return getTasksManagerImpl().listAllTasksForService(serviceId);
+	}
 
+	@Override
+	public List<Task> listAllTasksInState(PerunSession perunSession, Task.TaskStatus state) {
+		return getTasksManagerImpl().listAllTasksInState(state);
 	}
 
-	public ServicesManagerBl getServicesManagerBl() {
-		return servicesManagerBl;
+	@Override
+	public List<Task> listAllTasksNotInState(PerunSession sess, Task.TaskStatus state) {
+		return getTasksManagerImpl().listAllTasksNotInState(state);
 	}
 
+	@Override
+	public void removeAllTasksForService(PerunSession sess, Service service) {
+		for(Task task : listAllTasksForService(sess, service.getId())) {
+			getTasksManagerImpl().deleteTaskResults(task.getId());
+			removeTask(sess, task.getId());
+		}
+	}
 
-	public TasksManagerImplApi getTasksManagerImpl() {
-		return tasksManagerImpl;
+	@Override
+	public void removeTask(PerunSession sess, int id) {
+		getTasksManagerImpl().removeTask(id);
 	}
 
-	public void setServicesManagerBl(ServicesManagerBl servicesManagerBl) {
-		this.servicesManagerBl = servicesManagerBl;
+
+	@Override
+	public void removeTask(PerunSession sess, Service service, Facility facility) {
+		getTasksManagerImpl().removeTask(service, facility);
 	}
 
 	public void setPerunBl(PerunBlImpl perunBl) {
```

Comment:
Would be better to have all getters and setters before other methods (after the constructor).

Reply:
Done, and fixed the getters/setters names.